File: solutions/beecrowd/1022/1022.py
import sys


def gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return a


n = int(sys.stdin.readline())

for _ in range(n):
    a, _, b, o, c, _, d = sys.stdin.readline().split()
    a, b, c, d = map(int, [a, b, c, d])

    if o == '+':
        num = (b * d) * a // b + (b * d) * c // d
        den = b * d
    elif o == '-':
        num = (b * d) * a // b - (b * d) * c // d
        den = b * d
    elif o == '*':
        num = a * c
        den = b * d
    else:
        num = a * d
        den = b * c

    gcd_result = gcd(num, den)

    print(f'{num}/{den} = {num // gcd_result}/{den // gcd_result}')

File: solutions/beecrowd/1022/1022.clj
(ns main
  (:require [clojure.string :as str]))

(defn gcd [a b]
  (loop [a' a
         b' b]
    (if (= 0 b')
      a'
      (recur b' (mod a' b')))))

(defn solve-expression [num-1 den-1 operator num-2 den-2]
  (cond
    (= operator "+")
    [(+ (* num-1 den-2) (* num-2 den-1))
     (* den-1 den-2)]
    (= operator "-")
    [(- (* num-1 den-2) (* num-2 den-1))
     (* den-1 den-2)]
    (= operator "*")
    [(* num-1 num-2)
     (* den-1 den-2)]
    :else
    [(* num-1 den-2)
     (* num-2 den-1)]))

(defn main []
  (let [n (-> (read-line)
              (Integer/parseInt))]
    (loop [n' n]
      (when (> n' 0)
        (let [[num-1 _ den-1 operator num-2 _ den-2] (str/split (read-line) #" ")
              [num-1' den-1' num-2' den-2'] (map #(Integer/parseInt %) [num-1 den-1 num-2 den-2])
              [num den] (solve-expression num-1' den-1' operator num-2' den-2')
              gcd (gcd num den)]
          (println (format "%d/%d = %d/%d" num den (/ num gcd) (/ den gcd)))
          (recur (- n' 1)))))))

(main)
